priorityq/priorityq.go

```diff
@@ -0,0 +1,99 @@
+package priorityq
+
+import (
+	"container/heap"
+	"sync"
+
+	model "github.com/puchkovki/distr-model/model"
+)
+
+// An Item is something we manage in a priority queue.
+type Item struct {
+	value    string // The value of the item; arbitrary.
+	priority int    // The priority of the item in the queue.
+	// The index is needed by update and is maintained by the heap.Interface methods.
+	index int // The index of the item in the heap.
+}
+
+// A PriorityQueue implements heap.Interface and holds Items.
+type PriorityQueue []*Item
+
+// Len returns the PriorityQueue length
+func (pq PriorityQueue) Len() int {
+	return len(pq)
+}
+
+// Less returns is the j-th element priority of the pq PriorityQueue less than i-th element priority
+func (pq PriorityQueue) Less(i, j int) bool {
+	// We want Pop to give us the highest, not lowest, priority so we use "greater than" here.
+	return pq[i].priority > pq[j].priority
+}
+
+// Swap reverse j-th and i-th elements location in the pq PriorityQueue
+func (pq PriorityQueue) Swap(i, j int) {
+	pq[i], pq[j] = pq[j], pq[i]
+	pq[i].index = i
+	pq[j].index = j
+}
+
+// Push adds items to the PriorityQueue
+func (pq *PriorityQueue) Push(x interface{}) {
+	n := len(*pq)
+	item := x.(*Item)
+	item.index = n
+	*pq = append(*pq, item)
+}
+
+// Pop removes the highest-priority item from the PriorityQueue
+func (pq *PriorityQueue) Pop() interface{} {
```

Comment:
As I see, that's just a stack now. There isn't any item redistribution in the slice, so we just get **FILO**. And you don't use the `PriorityQueue` implementing the `heap.Interface`. Check `func main` (lines 77 and 84) in _godoc_ example for `container/heap`: https://play.golang.org/p/fa-K96WIgVS.